typespec-ccf/tsp-output/@azure-tools/typespec-autorest/0.0.1-preview/openapi.json

```diff
@@ -0,0 +1,923 @@
+          }
+        },
+        "consumes": [
+          "application/cose"
+        ]
+      }
+    },
+    "/gov/members/proposals/{proposalId}/ballots/{memberId}": {
+      "get": {
+        "operationId": "MemberGov_GetBallot",
+        "description": "Returns a member's submitted ballot.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/Azure.Core.Foundations.ApiVersionParameter"
+          },
+          {
+            "name": "proposalId",
+            "in": "path",
+            "required": true,
+            "description": "Unique ID assigned to this proposal at its submission, by the service.",
+            "format": "hex",
+            "maxLength": 64,
+            "type": "string"
+          },
+          {
+            "name": "memberId",
+            "in": "path",
+            "required": true,
+            "description": "ID of CCF member who signed and submitted this ballot.",
+            "format": "hex",
+            "maxLength": 64,
+            "type": "string"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The request has succeeded.",
+            "headers": {
+              "x-ms-ccf-transaction-id": {
+                "description": "Identifier for transaction this response was read from. The service state may be altered by other write requests. Reads at the same transactionId describe a consistent single point in the service history.",
+                "pattern": "^[0-9]+\\.[0-9]+$",
+                "type": "string"
+              }
+            },
+            "schema": {
+              "$ref": "#/definitions/Ballot"
+            }
+          },
+          "default": {
+            "description": "An unexpected error response.",
+            "headers": {
+              "x-ms-error-code": {
+                "description": "String error code indicating what went wrong.",
+                "type": "string"
+              }
+            },
+            "schema": {
+              "$ref": "#/definitions/Azure.Core.Foundations.ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/gov/members/proposals:propose": {
+      "post": {
+        "operationId": "MemberGov_Propose",
+        "description": "Submit a proposed change to the service. This will be assigned an ID by the service. Idempotence is based on the signed proposal body - submitting identical signed proposals multiple times will only create a single proposal on the server, unless sufficient time has passed (since the `created_at` signed COSE header), in which case the submission is stale and will return an error.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/Azure.Core.Foundations.ApiVersionParameter"
+          },
+          {
+            "name": "body",
+            "in": "body",
+            "required": true,
+            "description": "A raw CoseSign1 signature. The required protected headers and content schema will vary by action.",
```

Comment:
How do users know what the shape of the stuff in the COSE looks like? If it can't be documented in the Typespec, we should at least link to CCF documentation

Reply:
This is one of the big holes in this scheme, and something I will raise with the TypeSpec/Azure Core team. Afaict, there is no way to pass describe arbitrary metadata like this that will be handled by the emitters, other than verbose descriptions. Linking to the docs is one solution, but I'm hoping there's something I've missed that will let us document the payload scheme + required headers inline in the OpenAPI.